src/test/java/com/google/devtools/build/lib/authandtls/credentialhelper/GetCredentialsResponseTest.java

```diff
@@ -0,0 +1,74 @@
+package com.google.devtools.build.lib.authandtls.credentialhelper;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import com.google.gson.Gson;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for {@link GetCredentialsResponse}. */
+@RunWith(JUnit4.class)
+public class GetCredentialsResponseTest {
+  private static final Gson GSON = new Gson();
+
+  @Test
+  public void parseValid() {
+    assertThat(GSON.fromJson("{}", GetCredentialsResponse.class).getHeaders()).isEmpty();
+    assertThat(GSON.fromJson("{\"headers\": {}}", GetCredentialsResponse.class).getHeaders())
+        .isEmpty();
+
+    GetCredentialsResponse.Builder expectedResponseBuilder = GetCredentialsResponse.newBuilder();
+    expectedResponseBuilder.headersBuilder().put("a", ImmutableList.of());
+    expectedResponseBuilder.headersBuilder().put("b", ImmutableList.of("b"));
+    expectedResponseBuilder.headersBuilder().put("c", ImmutableList.of("c", "c"));
+    GetCredentialsResponse expectedResponse = expectedResponseBuilder.build();
+
+    assertThat(
+            GSON.fromJson(
+                "{\"headers\": {\"c\": [\"c\", \"c\"], \"a\": [], \"b\": [\"b\"]}}",
+                GetCredentialsResponse.class))
+        .isEqualTo(expectedResponse);
+  }
+
```

Comment:
Can you add tests for malformed `headers` fields, e.g.

`{"headers": null}`
`{"headers": "foo"}`
`{"headers": ["foo"]}`
`{"headers": {"foo": "bar"}}`